sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Group.java

```diff
@@ -0,0 +1,684 @@
+ *
+ * <p>When used without a combiner, this transforms simply acts as a {@link GroupByKey} but without
+ * the need for the user to explicitly extract the keys. For example, consider the following input
+ * type:
+ *
+ * <pre>{@code
+ * {@literal @DefaultSchema(JavaFieldSchema.class)}
+ * public class UserPurchase {
+ *   public String userId;
+ *   public String country;
+ *   public long cost;
+ *   public double transactionDuration;
+ * }
+ *
+ * {@literal PCollection<UserPurchase>} purchases = readUserPurchases();
+ * }</pre>
+ *
+ * <p>You can group all purchases by user and country as follows:
+ *
+ * <pre>{@code
+ * {@literal @DefaultSchema(JavaFieldSchema.class)}
+ * {@literal PCollection<KV<Row, Iterable<UserPurchase>>} byUser =
+ *   purchases.apply(Group.byFieldNames("userId', "country"));
+ * }</pre>
+ *
+ * <p>However often an aggregation of some form is desired. The builder methods inside the Group
+ * class allows building up separate aggregations for every field (or set of fields) on the input
+ * schema, and generating an output schema based on these aggregations. For example:
+ *
+ * <pre>{@code
+ * {@literal PCollection<KV<Row, Row>>} aggregated = purchases
+ *      .apply(Group.byFieldNames("userId', "country")
+ *          .aggregateField("cost", Sum.ofLongs(), "total_cost")
+ *          .aggregateField("cost", Top.<Long>largestLongsFn(10), "top_purchases")
+ *          .aggregateField("cost", ApproximateQuantilesCombineFn.create(21),
+ *            Field.of("transactionDurations", FieldType.array(FieldType.INT64)));
+ * }</pre>
+ *
+ * <p>The result will be a new row schema containing the fields total_cost, top_purchases, and
+ * transactionDurations, containing the sum of all purchases costs (for that user and country), the
+ * top ten purchases, and a histogram of transaction durations.
+ *
+ * <p>Note that usually the field type can be automatically inferred from the {@link CombineFn}
+ * passed in. However sometimes it cannot be inferred, due to Java type erasure, in which case a
+ * {@link Field} object containing the field type must be passed in. This is currently the case for
+ * ApproximateQuantilesCombineFn in the above example.
+ */
+@Experimental(Kind.SCHEMAS)
+public class Group {
+  /**
+   * Returns a transform that groups all elements in the input {@link PCollection}. The returned
+   * transform contains further builder methods to control how the grouping is done.
+   */
+  public static <T> Global<T> globally() {
+    return new Global<>();
+  }
+
+  /**
+   * Returns a transform that groups all elements in the input {@link PCollection} keyed by the list
+   * of fields specified. The output of this transform will be a {@link KV} keyed by a {@link Row}
+   * containing the specified extracted fields. The returned transform contains further builder
+   * methods to control how the grouping is done.
+   */
+  public static <T> ByFields<T> byFieldNames(String... fieldNames) {
```

Comment:
I think it needs an example here, the pseudocode of equivalent `GBK`

Reply:
Example already included in header JavaDoc.